File: backup_before_industrialisation_20250626/backend/metier_generator/templates/threed/tests_centralises/i18n/test_i18n.js
// Tests avancés pour l’i18n threed
const { translate, getAvailableLanguages } = require('../../api/i18n');
describe('i18n threed', () => {
  it('doit traduire une clé existante', () => {
    expect(translate('welcome', 'fr')).toBe('Bienvenue');
  });
  it('doit retourner la liste des langues disponibles', () => {
    const langs = getAvailableLanguages();
    expect(langs).toContain('fr');
    expect(langs).toContain('en');
  });
  it('doit retourner la clé si la traduction n’existe pas', () => {
    expect(translate('not_found', 'zz')).toBe('not_found');
  });
});

File: backend/metier_generator/templates/threed/tests_centralises/i18n/test_i18n.py
"""
Test centralisé i18n threed (Python).
"""

from ...i18n import translate, get_available_languages


def test_translate_existing_key():
    assert translate("welcome", "fr") == "Bienvenue"


def test_available_languages():
    langs = get_available_languages()
    assert "fr" in langs
    assert "en" in langs


def test_translate_missing_key():
    assert translate("not_found", "zz") == "not_found"
